test/e2e/framework.go

```diff
@@ -203,41 +204,46 @@ type PodIPs struct {
 	ipStrings []string
 }
 
-type deployAntreaOptions int
-
-const (
-	deployAntreaDefault deployAntreaOptions = iota
-	deployAntreaIPsec
-	deployAntreaWireGuardGo
-	deployAntreaCoverageOffset
-)
+type deployAntreaConfiguration struct {
+	name          string
+	ymls          []string
+	windowsAgents bool
+}
 
-func (o deployAntreaOptions) WithCoverage() deployAntreaOptions {
-	return o + deployAntreaCoverageOffset
+func (c deployAntreaConfiguration) withCoverage() deployAntreaConfiguration {
+	var covYMLMap = map[string]string{
+		antreaYML:            antreaCovYML,
+		antreaIPSecYML:       antreaIPSecCovYML,
+		antreaWireGuardGoYML: antreaWireGuardGoCovYML,
+	}
+	newConfig := deployAntreaConfiguration{
+		name: c.name,
+		ymls: []string{},
+	}
+	for i := range c.ymls {
+		yml := c.ymls[i]
+		if _, ok := covYMLMap[yml]; ok {
+			newConfig.ymls = append(newConfig.ymls, covYMLMap[yml])
+		} else {
+			newConfig.ymls = append(newConfig.ymls, yml)
+		}
+	}
+	return newConfig
 }
 
-func (o deployAntreaOptions) DeployYML() string {
-	return deployAntreaOptionsYML[o]
+func (c deployAntreaConfiguration) getYML() []string {
+	return c.ymls
 }
 
-func (o deployAntreaOptions) String() string {
-	return deployAntreaOptionsString[o]
+func (c deployAntreaConfiguration) withWindowsAgents() bool {
+	return c.windowsAgents
```

Comment:
I think the naming is a bit confusing here, as `withCoverage()` generates a new configuration where coverage has been enabled, while `withWindowsAgents()` does not generate a new configuration but instead returns true if the current configuration supports Windows.

I suggest renaming `withWindowsAgents` to `hasWindowsAgents`, or something similar

Reply:
Makes sense. Changed to `hasWindowsAgents`